### PR TITLE
add wireguard-tools

### PR DESCRIPTION
**Pull request by stephengroat**
#1739 #1742

### CODE REVIEW

File: packages/wireguard-tools/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+TERMUX_PKG_HOMEPAGE=https://www.wireguard.io
```
Comment (zx2c4) [inline packages/wireguard-tools/build.sh]: The correct URL ends in .com, not .io.

Comment (stephengroat) [reply packages/wireguard-tools/build.sh]: fixed, thanks for the catch

**Comment by stephengroat** [comment]
@zx2c4 thanks for the help, i was planning on waiting until `libmnl` get merged and then adding in the rest of the additions (since i know it's a dependency, i'm just being lazy waiting). i know this doesn't build yet, more of a marker

**Comment by zx2c4** [comment]
Now that @fornwall has merged the other stuff, I assume all you need to do is set the package directory, and then pass make those environment variables mentioned in the original issue.

**Comment by zx2c4** [comment]
WITH_BASHCOMPLETION=yes \
                WITH_WGQUICK=no \
                WITH_SYSTEMDUNITS=no

File: packages/wireguard-tools/build.sh [hunk 2]
```diff
@@ -5,11 +5,4 @@ TERMUX_PKG_SRCURL=https://git.zx2c4.com/WireGuard/snapshot/WireGuard-$TERMUX_PKG
...
 TERMUX_PKG_BUILD_IN_SRC=yes
-
-termux_step_make() {
-  make -C src/tools WITH_BASHCOMPLETION=yes WITH_WGQUICK=no WITH_SYSTEMDUNITS=no
-}
-
-termux_step_make_install() {
-  make -C src/tools install PREFIX=$TERMUX_PREFIX WITH_BASHCOMPLETION=yes WITH_WGQUICK=no WITH_SYSTEMDUNITS=no
-}
+TERMUX_PKG_EXTRA_MAKE_ARGS=" -C src/tools WITH_BASHCOMPLETION=yes WITH_WGQUICK=no WITH_SYSTEMDUNITS=no"
```
Comment (zx2c4) [inline packages/wireguard-tools/build.sh]: Is the space after " necessary?

Comment (stephengroat) [reply packages/wireguard-tools/build.sh]: in case other case get specified elsewhere, the space if good to make sure that concatenation goes smoothly

**Comment by stephengroat** [comment]
@fornwall anything else needed for this?

**Comment by zx2c4** [comment]
You probably want to include the android port of wg-quick as part of these tools, which is available here:

```
wget https://git.zx2c4.com/wireguard-android/plain/app/tools/wg-quick.c
gcc -o wg-quick ./wg-quick.c
```

**Comment by stephengroat** [comment]
@zx2c4 is that tool version controlled at all? seems like to might be better to package versions than just randomly pulling in a c file

**Comment by zx2c4** [comment]
It is version controlled. It's out of https://git.zx2c4.com/wireguard-android/ -- this repo. I'm making bump commits for when I increment the apk builds, but I'm not yet tagging. So, as of writing, this would be the file out of the latest bump commit:

https://git.zx2c4.com/wireguard-android/plain/app/tools/wg-quick.c?id=70a9e0d31b914c81b768a192315eae3de35e975d

Later on I'll probably start adding tags, for convenience though. For now, the above should suffice?

**Comment by Grimler91** [comment]
@zx2c4 I tried building 0.0.20180202 but it fails on aarch64&x86_64 with:
```
  CC      /home/builder/.termux-build/wireguard-tools/src/src/tools/setconf.o
curve25519.c:16:13: error: typedef redefinition with different types ('u64' (aka 'unsigned long') vs '__u64' (aka 'unsigned long long'))
typedef u64 __le64;
            ^
/home/builder/.termux-build/_lib/16-x86_64-21-v3/bin/../sysroot/usr/include/linux/types.h:30:25: note: previous definition is here
typedef __u64 __bitwise __le64;
                        ^
  CC      /home/builder/.termux-build/wireguard-tools/src/src/tools/showconf.o
  CC      /home/builder/.termux-build/wireguard-tools/src/src/tools/pubkey.o
1 error generated.
<builtin>: recipe for target 'curve25519.o' failed
make: *** [curve25519.o] Error 1
make: *** Waiting for unfinished jobs....
  CC      /home/builder/.termux-build/wireguard-tools/src/src/tools/genkey.o
make: Leaving directory '/home/builder/.termux-build/wireguard-tools/src/src/tools'
```
Originates in commit [6b28fa6](https://git.zx2c4.com/WireGuard/commit/src/tools/curve25519.c?id=6b28fa69ad38d1a183beb288a5547a3ca0b002ed) it seems, I added a small patch for affected arches for now.

-----------

@stephengroat I opened a pull request to your branch. Could you also rebase this branch? 

I also added wg-quick, but it needs some patching before it can work from termux (it seems), it
is unable to find any conf files and doesn't recognize any interfaces (even though it's running as root). 

I've made .debs with wg-quick available here: [aarch64](https://grimler.se/dists/testing/misc/binary-aarch64/wireguard-tools_0.0.20180202_aarch64.deb) [arm](https://grimler.se/dists/testing/misc/binary-arm/wireguard-tools_0.0.20180202_arm.deb) [i686](https://grimler.se/dists/testing/misc/binary-i686/wireguard-tools_0.0.20180202_i686.deb) [x86_64](https://grimler.se/dists/testing/misc/binary-x86_64/wireguard-tools_0.0.20180202_x86_64.deb) if anyone wants to test it.

Source tree is [here](https://github.com/Grimler91/termux-packages/tree/wireguard).

**Comment by zx2c4** [comment]
@Grimler91 Thanks for your work on this, and the useful report. I've addressed that [here](https://git.zx2c4.com/WireGuard/commit/?id=278502d4c4fd89b7cc864d437bc985a664579717), and this will be in the next snapshot revision.

Some nitpicks in your tree:

```
local wg_quick_commit=73431b5458ed352e49e6fefe85f075483035a1c9
```
The `local` keyword doesn't have any meaning in this context.

```

 #ifndef WG_CONFIG_SEARCH_PATHS
-#define WG_CONFIG_SEARCH_PATHS "/data/misc/wireguard /data/data/com.wireguard.android/files"
+#define WG_CONFIG_SEARCH_PATHS "@TERMUX_HOME@ @TERMUX_PREFIX@/etc/wireguard /data/misc/wireguard /data/data/com.wireguard.android/files"
 #endif
```

First, having `TERMUX_HOME` in there is just plain wrong. I'll fix that up in my suggestion below.

Secondly, and more importantly, the reason this is in an `ifndef` block is so that you don't have to patch the source but can rather address this from the `CFLAGS` environment variable or from the Makefile. For example, you could add something like this:

`'-DWG_CONFIG_SEARCH_PATHS="$(TERMUX_HOME)/.wireguard $(TERMUX_PREFIX)/etc/wireguard /data/misc/wireguard /data/data/com.wireguard.android/files"'`

(It's important that you keep the last two paths, as well.)

**Comment by Grimler91** [comment]
@zx2c4 Thanks for your feedback! 
I added another commit that cleaned it up as you suggested. 

debs are updated as well if anyone wants to install wireguard-tools before it is merged into the main repo.

**Comment by zx2c4** [comment]
@Grimler91 I made several comments on that commit. Let me know what you think.

**Comment by Grimler91** [comment]
@zx2c4 thanks! I've updated the branch again.

**Comment by Grimler91** [comment]
@stephengroat Maybe we should move this package to the root repo instead, wg might be a bit useful without root but wg-quick requires root. 
Most people interested in the package will have a custom kernel (with the wireguard module) and are most likely rooted.

**Comment by zx2c4** [comment]
No, please don't do that. `wg-quick` has in it explicit logic for non-root user. And `wg` has operations that are useful for non-root user as well, including some operations that don't even require the kernel module.

**Comment by fornwall** [comment]
This has now been merged in an updated PR https://github.com/termux/termux-packages/pull/2650 - thanks a lot for the PR @stephengroat and sorry for the delay!

**Pull request closed without merging**